ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HiveAugmentSnapshotMaterializationRule.java

```diff
@@ -125,11 +131,12 @@ public void onMatch(RelOptRuleCall call) {
     Table table = hiveTable.getHiveTableMD();
 
     SnapshotContext mvMetaTableSnapshot = mvMetaStoredSnapshot.get(table.getFullyQualifiedName());
-    if (mvMetaTableSnapshot.equals(table.getStorageHandler().getCurrentSnapshotContext(table))) {
+    if (Objects.equals(mvMetaTableSnapshot, table.getStorageHandler().getCurrentSnapshotContext(table))) {
```

Comment:
is there a null check for `table.getStorageHandler()`?

Reply:
Added null check.
In theory this can not happen because this rule is used only if MV base tables are Iceberg. Mixed mode is not supported.